### ISSUE TITLE
Adding / removing members from channel doesn't update members array in ChannelContext

### ISSUE THREAD

**Issue by veloware** [GetStream/stream-chat-react#538]
I am adding members (moderators) or removing members to/from a channel - 

_server side_ (add as moderator)

```
channel.addModerators(userIds)
```

or

_client side_ (remove member) using `withChannelContext`
```
props.channel.removeMembers([id])
```

however neither is causing `withChannelContext` to provide an updated array of members, using `props.members`.  Logging `props.members` shows only the initial array of members before adding or removing.  I understand there are a range of events that get fired for `channel`, e.g. `member.added`, however I would expect the `withChannelContext` to provide a new immutable array of members once members is updated?

gz#6223

**Comment by veloware** [comment]
Hi, is there an update on this? This is proving to be a poor user experience for us not having the array of members being updated.  Thanks

**Comment by jaapbakker88** [comment]
working on this here: #561

**Comment by jaapbakker88** [comment]
@veloware this should be resolved in version v3.0.1
